deliverylocation/tests/test_deliverylocation.py

```diff
@@ -0,0 +1,87 @@
+import pytest
+from deliverylocation.models import Supplier, DeliveryLocation
+from django.core.exceptions import ValidationError
+import datetime
+
+test_data = [
+                ('taxas', 'firstname', 'lastname', 'password', 'supplier', 'business name'),
+                ('ed', 'edd', 'eddy', 'jawbreaker101', 'supplier', 'jawbreaker inc'),
+        ]
+
+
+@pytest.fixture
+def supplier0():
+    return Supplier(test_data[0])
+
+
+@pytest.fixture
+def deliverylocation0(supplier0):
+    return DeliveryLocation(user_name=supplier0, location="Qiryat Shemona", date=datetime.date(2022, 12, 30))
+
+
+@pytest.fixture
+def deliverylocation1(supplier0):
+    return DeliveryLocation(user_name=supplier0, location="Haifa", date=datetime.date(2022, 12, 31))
+
+
+class TestDeliveryLocationModel:
+    @pytest.mark.django_db()
+    def test_add_delivery_location(self, supplier0, deliverylocation0, deliverylocation1):
+        supplier0.save()
+        deliverylocation0.add_delivery_location()
+        deliverylocation1.add_delivery_location()
+        assert deliverylocation0 in DeliveryLocation.objects.all()
+        assert deliverylocation1 in DeliveryLocation.objects.all()
+
+    @pytest.mark.django_db()
+    def test_delete_non_existing_delivery_location(self):
+        with pytest.raises(DeliveryLocation.DoesNotExist):
+            DeliveryLocation.remove_delivery_location_by_id(-1)
+
+    @pytest.mark.django_db()
+    def test_delete_delivery_location(self, supplier0, deliverylocation0):
+        supplier0.save()
+        deliverylocation0.add_delivery_location()
+        assert deliverylocation0 in DeliveryLocation.objects.all()
+        deliverylocation0.remove_delivery_location()
+        assert deliverylocation0 not in DeliveryLocation.objects.all()
+
+    @pytest.mark.django_db()
+    def test_filter_by_id(self, supplier0, deliverylocation0, deliverylocation1):
+        supplier0.save()
+        deliverylocation0.add_delivery_location()
+        deliverylocation1.add_delivery_location()
+        assert deliverylocation0 == DeliveryLocation.get_dilvery_location_by_id(deliverylocation0.id)
+        assert deliverylocation1 == DeliveryLocation.get_dilvery_location_by_id(deliverylocation1.id)
```

Comment:
What's the incentive for doing it twice? If it changes for some reason, you have to change it twice.

Reply:
I'd like to see as well a test in which the result is empty (same for other filters as well).

Reply:
Done